src/components/EmojiPicker/EmojiPicker.tsx

```diff
@@ -87,19 +88,23 @@ function EmojiPicker({viewportOffsetTop}: EmojiPickerProps, ref: ForwardedRef<Em
 
         const anchorOriginValue = anchorOrigin ?? DEFAULT_ANCHOR_ORIGIN;
 
+        // It's possible that the anchor is inside an active modal (e.g., add emoji reaction in report context menu).
+        // So, we need to get the anchor position first before closing the active modal which will also destroy the anchor.
         calculateAnchorPosition(emojiPopoverAnchor?.current, anchorOriginValue).then((value) => {
-            onWillShow?.();
-            setIsEmojiPickerVisible(true);
-            setEmojiPopoverAnchorPosition({
-                horizontal: value.horizontal,
-                vertical: value.vertical,
+            Modal.close(() => {
```

Comment:
one more case, we should have handled keyboard close 
https://github.com/Expensify/App/issues/65341#issuecomment-3027948541